### ISSUE TITLE
Improve group management

### ISSUE THREAD

**Issue by ogarcia** [nextcloud/server#5704]
At this moment the group management is a little hell. Groups are sorted by number of members instead alphabetically, cannot rename groups, etc. I think that this must be improved at least with:

- [x] Sort groups alphabetically
- [x] Rename groups
- [ ] Bulk add group members
- [ ] Bulk remove group members (https://github.com/nextcloud/server/issues/384)

**Comment by detm54** [comment]
A small button/switch to sort groups by either number of members, creation date (age) or alphabetically would be pretty awesome. Especially a-z to match the order of grouplist when assigning groups for new users.

Have a nice day!

**Comment by MorrisJobke** [comment]
See also #384

**Comment by bpcurse** [comment]
The side-menu's sorting by user count shouldn't be the default view. Especially when you have created groups in a logical naming concept, it seems wiser to display them by default sorted by name and let the user choose another view if necessary.

**Comment by DJCrashdummy** [comment]
IMHO these would be very nice features to have. :+1: but the **most important feature is renaming groups**, since all other can somehow be accomplished or worked around...

so is there any way to rename a group, until this gets implemented?

**Comment by Dubidubiduu** [comment]
Yes, renaming groups is very important to me too, cuz I don't want to
.. create a group
.. give access to the new group
.. add all the members to the new group
.. and then delete the old group

**Comment by DJCrashdummy** [comment]
@Dubidubiduu well, this would be a tedious but still practicable workaround... but i'm **concerned about the existing group-shares** because **with this workaround they will be lost**!

and for this case (keeping group-shares) there is no workaround i know; hence the question, if there is any chance to rename a group for now (via manipulating the database, etc.).

**Comment by fredt34** [comment]
C'mon, how can it be soooo difficult to rename a group??

**Comment by DjSni** [comment]
For the group administrator, it should also be possible to add existing users to his group(s).

**Comment by fenvarien** [comment]
**Renaming groups** is still a much needed feature. Is it currently in development? I'm an admin for a nextcloud installation in a school. We need the ability to rename a class from eg "7a" to "8a" with the beginning of thr next school year. But we want to keep the students in that group/class.

**Comment by SimJoSt** [comment]
@fenvarien there is a separate issue for this: https://github.com/nextcloud/server/issues/9209 :)

**Comment by blackerking** [comment]
At least the sorting function would be an useful utility.

**Comment by cl4x** [comment]
honestly it's really unusual that you can't rename simple group names... sometimes the project focuses on new big features but gets lost in trivial old things.

**Comment by stefanp2021** [comment]
Hi! I'd like to push this issue! I support it, and would greatly appreciate it!

**Comment by kleener** [comment]
What is the status here?
Can not imagine that it is all too difficult?

**Comment by LostinSpacetime** [comment]
We can rename Groups in Nextcloud 25 :)

**Comment by kleener** [comment]
Now if you tell me that sorting also works, I'll be happy.

**Comment by susnux** [comment]
Just for information, sorting by

> creation date (age)

requires backend changes as the current interface does not provide such information.

**Comment by David-Rodriguez-cabrera** [comment]
the Bulk add group members options are very necessary

**Comment by ThaDaVos** [comment]
What's the status on the Group Sorting, all issues lead to this one but last update is from last year

**Comment by susnux** [comment]
> What's the status on the Group Sorting, all issues lead to this one but last update is from last year

This is already possible (since ~7 years) using the system config `sort_groups_by_name`
Ref: https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/config_sample_php_parameters.html#sort-groups-by-name

**Comment by ThaDaVos** [comment]
@susnux  - that has done it, thanks

**Comment by susnux** [comment]
In 29 there will be also an option to sort it on the frontend without loosing the member count.

**Comment by tatrapikao** [comment]
I would suggest the solution that contacts app provides. every group has a context menu wich provides the option "add members" that spawns a window with a search bar that allows adding multiple users to the given group. As we speak about contact's app - it would be realy enjoyable to provide better integration with contact groups - as suggested here: https://github.com/nextcloud/contacts/issues/3835

**Comment by Antreesy** [comment]
UI examples for bulk add from other apps (Not sure if this would fit for bulk deletion):

Contacts (Teams) | Talk
-- | --
<img width="416" height="335" alt="Image" src="https://github.com/user-attachments/assets/c7ef96b7-b83d-4fee-bc99-2cd29650962e" />  |  <img width="423" height="462" alt="Image" src="https://github.com/user-attachments/assets/7709514d-fb3a-4f6a-995a-2ec2d15f0620" />

IMO Files app selection also fits here for both adding/deleting (with additional confirmation dialog afterwards):

<img width="430" height="153" alt="Image" src="https://github.com/user-attachments/assets/212c18c6-868f-4fbc-9e7a-05d5ac754019" />

**Comment by Clementine46** [comment]
Here are some propositions that were discussed with the Nextcloud design team to improve the editing of an account.

Rather than editing directly in an account line I suggest having a pop up just like the one for creating a new account: 

<img width="5760" height="3048" alt="Image" src="https://github.com/user-attachments/assets/0abbbabb-cd0e-4b35-85ad-e0d8832a5d2e" />

One of the main issue in my opinion is for options with multiple choices like "Groups", "Group Admin for" and "Manager" because they have 3 functions in 1 : research (typing), results (suggested groups) and selections (selected groups).

It's too many functions for one field and as a result, it's unclear that it is possible to type in the field once some groups have been selected + the user has to scroll horizontally to see all the groups.

<img width="303" height="98" alt="Image" src="https://github.com/user-attachments/assets/94a36125-92de-4d3d-acc8-a78dcf7a0ac8" />

With a separate pop up window to edit an account, we could have the selected groups under the research field:

<img width="5760" height="3048" alt="Image" src="https://github.com/user-attachments/assets/a8264cd1-a103-4134-b83e-17669d756e51" />

That way the research field can open on top of it:

<img width="5760" height="3048" alt="Image" src="https://github.com/user-attachments/assets/bfa6a08a-e97c-45e7-9b62-7edb51802206" />

<img width="5760" height="3048" alt="Image" src="https://github.com/user-attachments/assets/c21aa508-378a-4082-b021-f0731f8effc5" />

<img width="5760" height="3048" alt="Image" src="https://github.com/user-attachments/assets/944a491f-e387-482f-8588-2af9b95585c6" />

I also suggest removing the password from the main page to gain space since the column always appear empty and keeping the display name and account name in a singular column (I believe it used to be like that) would also help gain some space:

<img width="5760" height="3048" alt="Image" src="https://github.com/user-attachments/assets/9d51e073-20ea-4fc3-a2b0-7a05d88d3395" />

@nimishavijay @marcoambrosini @jancborchardt what do you think?

**Comment by tatrapikao** [comment]
Are there any thopughts towards bulk-editing and multiselect?

**Comment by pringelmann** [comment]
@Clementine46 Thanks for the detailed proposal and mockups! 

The separation of the search field from selected items makes a lot of sense - the current inline tags inside the input are hard to discover and interact with, especially once several groups are selected and horizontal scrolling kicks in.

From a component perspective, the pattern you're proposing (search field + tags displayed beneath) isn't currently available as a single component in `nextcloud-vue`. It can be composed using `NcSelect` (in single-select mode as the search/picker) combined with `NcChip` components for the selected items underneath. This is a straightforward composition, but if we expect this pattern to appear in more places, it might be worth discussing whether it should become a dedicated component in the library.

A simpler alternative is to prevent it from wrapping:

<img width="724" height="361" alt="Image" src="https://github.com/user-attachments/assets/86867c5b-210d-4ee7-8a02-83954b0b07b0" />

**Comment by susnux** [comment]
> From a component perspective, the pattern you're proposing (search field + tags displayed beneath) isn't currently available as a single component in nextcloud-vue.

@ShGKme is already planning to create such a component (similar to github labels) which should be reused here once available.
